### PR TITLE
Docs/ad4m model api refactor

### PR DESCRIPTION
**Pull request by jhweir**
# PR: Update documentation for Ad4mModel API refactor

**Branch:** `docs/ad4m-model-api-refactor`
**Files changed:** 6 | **+1,060 / −944 lines**

## Summary

Updates all developer-facing documentation to reflect the Ad4mModel decorator API refactoring (PR #717). Covers API renames, structural improvements, accuracy fixes, and new feature documentation.

### 1. `docs: update documentation for Ad4mModel API refactor`
Systematic rename pass across all 6 doc files:
- `@Subject` → `@Model`, `@SubjectProperty` → `@Property`, `@SubjectCollection` → `@HasMany`, `@SubjectFlag` → `@Flag`
- `SubjectEntity` → `Ad4mModel`, `.baseExpression` → `.id`
- `SDNAClass.generateSDNA()` → `Ad4mModel.generateSDNA()`
- Updated import paths, constructor signatures, query builder methods
- Fixed getter syntax errors (`FROM $parent` → raw SurrealQL expression)

**Files:** model-classes.mdx, surreal-queries.mdx, batch-operations.mdx, hooks.mdx, index.mdx, social-dna.mdx

### 2. `docs: restructure batch-operations, model-classes, and surreal-queries guides`
Structural improvements for clarity and scannability:
- **batch-operations.mdx** — Promoted `transaction()` as the primary pattern; demoted manual `createBatch()`/`commitBatch()` to an "Advanced" section
- **model-classes.mdx** — Moved "How It Works Under the Hood" earlier in the page; condensed the JSON Schema section from ~330 lines to ~135 using a resolution cascade table
- **surreal-queries.mdx** — Merged two overlapping graph traversal sections into one progressive section; fixed inaccurate index claims against actual `DEFINE INDEX` statements in source

### 3. `docs: add missing API documentation to model-classes guide`
Filled documentation gaps found by auditing the source code:
- **Built-in instance properties** — `.author`, `.createdAt`, `.updatedAt` (with usage in `order` queries)
- **RelationSubQuery** — Filtering, sorting, and paginating eager-loaded relations via `include`
- **Full PropertyOptions table** — `transform`, `local`, `initial` (sentinel behavior), `getter` (Base placeholder, SurrealQL wrapping)
- **`@Optional` decorator** — Mentioned as explicit alias for `@Property` with `required: false`
- **Default `through`** — Documented that omitting `through` on relations defaults to `'ad4m://has_child'`
- **Introspection** — `getModelMetadata()`, `generateSDNA()`, `generateSHACL()` with usage example

## Files changed

| File | Summary |
|------|---------|
| `model-classes.mdx` | Full rewrite + new sections (1,103 lines) |
| `surreal-queries.mdx` | Merged graph traversal, fixed index docs (657 lines) |
| `batch-operations.mdx` | Restructured around transaction() (227 lines) |
| `hooks.mdx` | API renames only |
| `index.mdx` | API renames, fixed getter syntax |
| `social-dna.mdx` | API renames only |

## Summary by CodeRabbit

* **Documentation**
  * Rewrote developer guides to promote a transaction-first workflow with updated examples, best practices, migration notes, and an advanced manual-batch section.
* **New Patterns**
  * Transactional API introduced as the recommended automatic commit/rollback pattern; manual batch control retained as an advanced option.
* **API & Examples**
  * Hook replaced by a live/reactive query API (returns "data", supports single-instance mode); model decorators and relation patterns simplified with updated examples and usage.
* **SurrealQL**
  * Query guide reorganized with clearer fields/indexes, traversal guidance, and updated result aliasing.

### CODE REVIEW

File: docs-src/pages/developer-guides/model-classes.mdx [hunk 1]
```diff
@@ -70,128 +81,208 @@ recipe.difficulty = 3;
...
-    through: "recipe://description",
-    resolveLanguage: "literal"
-  })
+  // Optional property — same behavior
+  @Property({ through: "recipe://description" })
   description?: string;
+
+  // Required property — a placeholder link is created on save
+  @Property({ through: "recipe://category", required: true })
+  category: string = "";
```
[the remark this answers is not in the document]
Comment (jhweir) [reply docs-src/pages/developer-guides/model-classes.mdx:114]: The docs are correct on this one:

The public `Property` decorator (`core/src/model/decorators.ts` lines 1033–1038) wraps the internal `applyPropertyMetadata` and **fills in the sentinel before the check runs**:

```typescript
export function Property(opts: PropertyOptions) {
    const required = opts.required ?? false;
    return applyPropertyMetadata({
        ...opts,
        required,
        initial: opts.initial ?? (required ? "literal://string:uninitialized" : undefined),
    });
}
```

So calling `@Property({ through: "recipe://category", required: true })` **without** an explicit `initial`:

1. `Property` sets `initial` to `"literal://string:uninitialized"` automatically
2. `applyPropertyMetadata` then sees `opts.initial` is truthy → the `if (opts.required && !opts.initial)` check passes → no throw

The `applyPropertyMetadata` guard at `decorators.ts` line 315 only catches direct **internal** callers that bypass the `Property` wrapper. The public API never throws here.

The docs at `model-classes.mdx` line 113 (`@Property({ through: "recipe://category", required: true })`) and the table at line 143 (describing the sentinel behavior) both correctly describe this behavior.

File: docs-src/pages/developer-guides/batch-operations.mdx [hunk 2]
```diff
@@ -135,32 +178,15 @@ await perspective.commitBatch(batchId);
...
-   - Test both successful and failed batch scenarios
-   - Verify data consistency after batch operations
-   - Include edge cases in your test suite
+1. **Prefer `transaction()`**: Use `Ad4mModel.transaction()` over manual `createBatch()`/`commitBatch()` — if the callback throws, the batch is not committed and no changes are persisted.
+
+2. **Batch Size**: Keep batches focused on related operations. Avoid extremely large batches that could impact memory usage — split very large operations into multiple smaller batches.
+
+3. **Error Handling**: With `transaction()`, errors thrown inside the callback automatically prevent commit. With manual batches, wrap operations in try/catch.
+
+4. **Resource Management**: Don't keep batches open longer than necessary. With `transaction()` this is handled for you.
```
[the remark this answers is not in the document]
Comment (jhweir) [reply docs-src/pages/developer-guides/batch-operations.mdx:187]: **No fix needed** — the docs already accurately reflect the current behavior after the rollback fixes from commit `1da33767`.

#### Current doc state

- **Line 181**: "if the callback throws, the batch is not committed and no changes are persisted" — **accurate**. `transaction()` calls `fn()` before `commitBatch()`, so if `fn` throws, `commitBatch` is never reached.
- **Line 185**: "errors thrown inside the callback automatically prevent commit" — **accurate**. Same reasoning: the exception propagates before `commitBatch` runs.
- **Line 219**: The migration example shows `transaction()` without try/catch — **accurate**. The batch simply won't commit if `fn` throws; the error propagates to the caller.

#### Analysis

The finding asks us to "explicitly state... no automatic rollback/cleanup" and "they must handle try/catch/cleanup for errors." But looking at the source code (`Ad4mModel.ts` lines 3594–3598):

```typescript
static async transaction<R = void>(
  perspective: PerspectiveProxy,
  fn: (tx: { batchId: string }) => Promise<R>,
): Promise<R> {
  const batchId = await perspective.createBatch();
  const result = await fn({ batchId });
  await perspective.commitBatch(batchId);
  return result;
}
```

There is nothing to clean up — an uncommitted batch is just an in-memory buffer that gets garbage-collected. There is no `cancelBatch()` or `deleteBatch()` API. The batch simply evaporates when it's never committed.

The docs correctly say "the batch is not committed and no changes are persisted" — that *is* the cleanup behavior. Adding warnings about manual cleanup would be misleading since there's nothing to manually clean up.

**Review by lucksus — APPROVED** [review]
